### PR TITLE
nc:myspot_gov normalize

### LINKED ISSUE #88: normalize myspot_gov from North Carolina
[learn our pipeline: normalize](https://github.com/CAVaccineInventory/vaccine-feed-ingest/wiki/Runner-pipeline-stages#normalize)

Blocked by #89 

Normalize existing [`.ndjson`](http://ndjson.org/) records into [the Vaccinate The States data schema](https://github.com/CAVaccineInventory/vaccine-feed-ingest/blob/main/vaccine_feed_ingest/schema/schema.py).

Read all files in the directory passed as the second argument (`sys.argv[2]`), transform each record into the normalized schema (filling in as much detail as possible), then output them to new files in the directory passed as the first argument (`sys.argv[1]`).

Check the wiki to learn more about the purpose of the normalize stage and how to get set up for development!

### Tips

1. Fetch and parse data for this site before you start developing:
    ```sh
    poetry run vaccine-feed-ingest fetch nc/myspot_gov
    poetry run vaccine-feed-ingest parse nc/myspot_gov
    ```

1. While working on your code, run it at any point:
    ```sh
    poetry run vaccine-feed-ingest normalize nc/myspot_gov
    ```

### Example
[Parsed files for `ak/arcgis`](https://github.com/CAVaccineInventory/vaccine-feed-ingest-results/tree/main/ak/arcgis/parsed) are converted to [normalized files](https://github.com/CAVaccineInventory/vaccine-feed-ingest-results/tree/main/ak/arcgis/normalized)

### PR DESCRIPTION
**Pull request by rossry**
# normalize myspot_gov for nc

| Key Details |
|-|
| Resolves #88 |
State: nc |
Site: myspot_gov |

## Notes
* ~~I borked my git branches, so this one also includes il/sfsites/parse.yml, which was PR'd in #213. Probably just wait for that one to be accepted before considering this one. (Or someone more fluent in git than me can try to pull the il/sfsites/parse.yml out of this one, so they're properly independent.)~~
* Uses "walmart" as the enum for Walmart; will add to wiki separately.

## Data sample

```json
{"id": "nc:myspot_gov:E-0000015", "name": "Kernersville Pharmacy", "address": {"street1": "841 Old Winston Road", "street2": "suite 90", "city": "Kernersville", "state": "NC", "zip": "27284"}, "location": {"latitude": 36.1122068, "longitude": -80.0905011}, "contact": [{"contact_type": null, "phone": "(336) 497-4511", "website": "https://kernersvillepharmacy.com/", "email": null, "other": null}], "languages": null, "opening_dates": [{"opens": "2021-03-15", "closes": null}], "opening_hours": null, "availability": null, "inventory": [], "access": null, "parent_organization": {"id": null, "name": "Kernersville Pharmacy LLC"}, "links": null, "notes": ["cvms_scheduling:F", "county:Forsyth"], "active": null, "source": {"source": "nc:myspot_gov", "id": "nc:myspot_gov:E-0000015", "fetched_from_uri": "https://myspot.nc.gov/api/get-vaccine-locations", "fetched_at": "2021-04-26T09:26:29.385831", "published_at": null, "data": {"City": "Kernersville", "latitude": "36.1122068", "Postal Code": 27284, "State": "NC", "Start Date": "20210315 000000.000", "longitude": "-80.0905011", "Moderna": "", "Event Type": "Not Applicable", "Provider Location Name": "Kernersville Pharmacy", "Event Location Id": "E-0000015", "Salesforce Id": "a2qt000000lSlkLAAS", "Janssen": "", "Appointments": "Yes, Both", "Web Address": "https://kernersvillepharmacy.com/", "Organization Name": "Kernersville Pharmacy LLC", "Event Location": "", "Pfizer": "", "Street Address": "841 Old Winston Road", "County": "Forsyth", "CVMS Scheduling": "F", "Appointment Phone": "(336) 497-4511", "Display to the Public": "Y", "Street Address 2": "suite 90", "End Date": ""}}}
{"id": "nc:myspot_gov:E-0000357", "name": "Advance Community Health", "address": {"street1": "1011 Rock Quarry Road", "street2": "", "city": "Raleigh", "state": "NC", "zip": "27610"}, "location": {"latitude": 35.7670203, "longitude": -78.6157125}, "contact": [{"contact_type": null, "phone": "919-831-3111", "website": "http://www.advancechc.org/vaccine/", "email": null, "other": null}], "languages": null, "opening_dates": [{"opens": "2021-03-15", "closes": null}], "opening_hours": null, "availability": null, "inventory": [{"vaccine": "moderna", "supply_level": "in_stock"}, {"vaccine": "pfizer_biontech", "supply_level": "out_of_stock"}, {"vaccine": "johnson_johnson_janssen", "supply_level": "out_of_stock"}], "access": null, "parent_organization": {"id": null, "name": "Advance Community Health, Inc."}, "links": null, "notes": ["cvms_scheduling:F", "county:Wake"], "active": null, "source": {"source": "nc:myspot_gov", "id": "nc:myspot_gov:E-0000357", "fetched_from_uri": "https://myspot.nc.gov/api/get-vaccine-locations", "fetched_at": "2021-04-26T09:26:29.385831", "published_at": null, "data": {"Pfizer": "N", "Appointments": "No, not taking appointments or walk-ins", "Event Type": "Not Applicable", "Street Address": "1011 Rock Quarry Road", "Start Date": "20210315 000000.000", "Organization Name": "Advance Community Health, Inc.", "Event Location": "", "CVMS Scheduling": "F", "State": "NC", "Janssen": "N", "Street Address 2": "", "End Date": "", "Event Location Id": "E-0000357", "Postal Code": "27610", "Web Address": "http://www.advancechc.org/vaccine/", "Provider Location Name": "Advance Community Health", "Salesforce Id": "a2qt000000lSlqQAAS", "Appointment Phone": "919-831-3111", "County": "Wake", "Display to the Public": "Y", "City": "Raleigh", "Moderna": "Y", "latitude": "35.7670203", "longitude": "-78.6157125"}}}
{"id": "nc:myspot_gov:E-0000358", "name": "Alamance County Health Department", "address": {"street1": "319 N Graham Hopedale Rd Suite B", "street2": "", "city": "Burlington", "state": "NC", "zip": "27217"}, "location": {"latitude": 36.1004866, "longitude": -79.4052922}, "contact": [{"contact_type": null, "phone": "336-290-0650", "website": "https://www.alamance-nc.com/healthdept/", "email": null, "other": null}], "languages": null, "opening_dates": [{"opens": "2021-03-15", "closes": null}], "opening_hours": null, "availability": null, "inventory": [{"vaccine": "moderna", "supply_level": "in_stock"}, {"vaccine": "pfizer_biontech", "supply_level": "in_stock"}, {"vaccine": "johnson_johnson_janssen", "supply_level": "out_of_stock"}], "access": null, "parent_organization": {"id": null, "name": "Alamance County Health Department"}, "links": null, "notes": ["cvms_scheduling:F", "county:Alamance"], "active": null, "source": {"source": "nc:myspot_gov", "id": "nc:myspot_gov:E-0000358", "fetched_from_uri": "https://myspot.nc.gov/api/get-vaccine-locations", "fetched_at": "2021-04-26T09:26:29.385831", "published_at": null, "data": {"Event Type": "Not Applicable", "longitude": "-79.4052922", "Street Address": "319 N Graham Hopedale Rd Suite B", "City": "Burlington", "Display to the Public": "Y", "Web Address": "https://www.alamance-nc.com/healthdept/", "latitude": "36.1004866", "Event Location": "", "Moderna": "Y", "Pfizer": "Y", "State": "NC", "Start Date": "20210315 000000.000", "CVMS Scheduling": "F", "County": "Alamance", "Postal Code": "27217", "End Date": "", "Janssen": "N", "Organization Name": "Alamance County Health Department", "Appointments": "Yes, Appointments Only", "Salesforce Id": "a2qt000000lSlqRAAS", "Appointment Phone": "336-290-0650", "Event Location Id": "E-0000358", "Street Address 2": "", "Provider Location Name": "Alamance County Health Department"}}}
{"id": "nc:myspot_gov:E-0000359", "name": "Chowan County Health Department", "address": {"street1": "202 West Hicks St", "street2": "", "city": "Edenton", "state": "NC", "zip": "27932"}, "location": {"latitude": 36.068496, "longitude": -76.609741}, "contact": [{"contact_type": null, "phone": "(252) 482-6003", "website": "http://www.arhs-nc.org/", "email": null, "other": null}], "languages": null, "opening_dates": [{"opens": "2021-03-15", "closes": null}], "opening_hours": null, "availability": null, "inventory": [{"vaccine": "moderna", "supply_level": "in_stock"}, {"vaccine": "pfizer_biontech", "supply_level": "out_of_stock"}, {"vaccine": "johnson_johnson_janssen", "supply_level": "out_of_stock"}], "access": null, "parent_organization": {"id": null, "name": "Albemarle Regional Health Services"}, "links": null, "notes": ["cvms_scheduling:F", "county:Chowan"], "active": null, "source": {"source": "nc:myspot_gov", "id": "nc:myspot_gov:E-0000359", "fetched_from_uri": "https://myspot.nc.gov/api/get-vaccine-locations", "fetched_at": "2021-04-26T09:26:29.385831", "published_at": null, "data": {"Appointments": "Yes, Walk-In Only", "Salesforce Id": "a2qt000000lSlqSAAS", "Provider Location Name": "Chowan County Health Department", "End Date": "", "Street Address 2": "", "latitude": "36.0684960", "Janssen": "N", "Event Location": "", "Moderna": "Y", "Organization Name": "Albemarle Regional Health Services", "Street Address": "202 West Hicks St", "Pfizer": "N", "longitude": "-76.6097410", "Event Type": "Not Applicable", "Postal Code": "27932", "Web Address": "http://www.arhs-nc.org/", "Event Location Id": "E-0000359", "City": "Edenton", "Appointment Phone": "(252) 482-6003", "County": "Chowan", "CVMS Scheduling": "F", "Start Date": "20210315 000000.000", "State": "NC", "Display to the Public": "Y"}}}
{"id": "nc:myspot_gov:E-0000360", "name": "Bertie County Health Department", "address": {"street1": "102 Rhodes Avenue", "street2": "", "city": "Windsor", "state": "NC", "zip": "27983"}, "location": {"latitude": 35.9891429, "longitude": -76.9301032}, "contact": [{"contact_type": null, "phone": "252-794-5322", "website": "https://www.arhs-nc.org/", "email": null, "other": null}], "languages": null, "opening_dates": [{"opens": "2021-03-15", "closes": null}], "opening_hours": null, "availability": null, "inventory": [{"vaccine": "moderna", "supply_level": "in_stock"}, {"vaccine": "pfizer_biontech", "supply_level": "out_of_stock"}, {"vaccine": "johnson_johnson_janssen", "supply_level": "out_of_stock"}], "access": null, "parent_organization": {"id": null, "name": "Albemarle Regional Health Services"}, "links": null, "notes": ["cvms_scheduling:F", "county:Bertie"], "active": null, "source": {"source": "nc:myspot_gov", "id": "nc:myspot_gov:E-0000360", "fetched_from_uri": "https://myspot.nc.gov/api/get-vaccine-locations", "fetched_at": "2021-04-26T09:26:29.385831", "published_at": null, "data": {"Event Location Id": "E-0000360", "Appointment Phone": "252-794-5322", "Display to the Public": "Y", "Salesforce Id": "a2qt000000lSlqTAAS", "Start Date": "20210315 000000.000", "Postal Code": "27983", "Web Address": "https://www.arhs-nc.org/", "Event Location": "", "latitude": "35.9891429", "End Date": "", "longitude": "-76.9301032", "State": "NC", "Street Address 2": "", "City": "Windsor", "County": "Bertie", "Appointments": "Yes, Walk-In Only", "Moderna": "Y", "Provider Location Name": "Bertie County Health Department", "Janssen": "N", "Street Address": "102 Rhodes Avenue", "Event Type": "Not Applicable", "Organization Name": "Albemarle Regional Health Services", "Pfizer": "N", "CVMS Scheduling": "F"}}}
{"id": "nc:myspot_gov:E-0000361", "name": "Gates County Health Department", "address": {"street1": "29 Medical Center Drive", "street2": "", "city": "Gates", "state": "NC", "zip": "27937"}, "location": {"latitude": 36.4525909, "longitude": -76.6999053}, "contact": [{"contact_type": null, "phone": "252-357-1380", "website": "http://www.arhs-nc.org/", "email": null, "other": null}], "languages": null, "opening_dates": [{"opens": "2021-03-15", "closes": null}], "opening_hours": null, "availability": null, "inventory": [{"vaccine": "moderna", "supply_level": "in_stock"}, {"vaccine": "pfizer_biontech", "supply_level": "out_of_stock"}, {"vaccine": "johnson_johnson_janssen", "supply_level": "out_of_stock"}], "access": null, "parent_organization": {"id": null, "name": "Albemarle Regional Health Services"}, "links": null, "notes": ["cvms_scheduling:F", "county:Gates"], "active": null, "source": {"source": "nc:myspot_gov", "id": "nc:myspot_gov:E-0000361", "fetched_from_uri": "https://myspot.nc.gov/api/get-vaccine-locations", "fetched_at": "2021-04-26T09:26:29.385831", "published_at": null, "data": {"State": "NC", "Web Address": "http://www.arhs-nc.org/", "CVMS Scheduling": "F", "Start Date": "20210315 000000.000", "Provider Location Name": "Gates County Health Department", "City": "Gates", "Moderna": "Y", "Event Location Id": "E-0000361", "Display to the Public": "Y", "longitude": "-76.6999053", "Street Address": "29 Medical Center Drive", "End Date": "", "Appointment Phone": "252-357-1380", "Street Address 2": "", "latitude": "36.4525909", "Appointments": "Yes, Walk-In Only", "Event Location": "", "Pfizer": "N", "Postal Code": "27937", "Event Type": "Not Applicable", "Janssen": "N", "County": "Gates", "Organization Name": "Albemarle Regional Health Services", "Salesforce Id": "a2qt000000lSlqUAAS"}}}
```

## Before Opening a PR
- [x] I tested this using the CLI (e.g., `poetry run vaccine-feed-ingest <state>/<site>`)
- [x] I ran auto-formatting:
  - [x] `poetry run black vaccine_feed_ingest/runners`
  - [x] `poetry run isort vaccine_feed_ingest/runners`

### CODE REVIEW

File: vaccine_feed_ingest/runners/nc/myspot_gov/normalize.py [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+        longitude=float(site["longitude"]),
+    )
+
+
+def _get_contacts(site: dict):
+    return [
+        schema.Contact(
+            phone=site["Appointment Phone"],
+            website=site["Web Address"],
+        )
```
Comment (eliblock) [inline vaccine_feed_ingest/runners/nc/myspot_gov/normalize.py:47]: Could you please make the following tweaks:
* add a _separate_ contact entry for each type (so if a location has both a website and a phone, they should have _two_ contact records).
* Format the phone number like `(###) ###-####` ([see schema](https://github.com/CAVaccineInventory/vaccine-feed-ingest/wiki/Normalized-Location-Schema#specification))

File: vaccine_feed_ingest/runners/nc/myspot_gov/normalize.py [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+        ret.append("event_type:" + site["Event Type"])
+    return ret
+
+
+def _get_source(site: dict, timestamp: str) -> schema.Source:
+    return schema.Source(
+        data=site,
+        fetched_at=timestamp,
+        fetched_from_uri="https://myspot.nc.gov/api/get-vaccine-locations",
+        id=_get_id(site),
```
Comment (eliblock) [inline vaccine_feed_ingest/runners/nc/myspot_gov/normalize.py:115]: Within the `Source` block, let's use the `id` defined in the data source directly ([see schema](https://github.com/CAVaccineInventory/vaccine-feed-ingest/wiki/Normalized-Location-Schema#specification)).

The "decorated" id which includes the file info is great for the general id of the normalized record!

File: vaccine_feed_ingest/runners/nc/myspot_gov/normalize.py [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+    if site["Organization Name"] == "":
+        return None
+    if site["Organization Name"] == "Walmart, Inc.":
+        return schema.Organization(name=site["Organization Name"], id="walmart")
+    return schema.Organization(name=site["Organization Name"])
+
+
+def _get_notes(site: dict):
+    ret = []
+    ret.append("cvms_scheduling:" + site["CVMS Scheduling"])
```
Comment (eliblock) [inline vaccine_feed_ingest/runners/nc/myspot_gov/normalize.py:103]: Did you happen to find out what `cvms_scheduling` means for this dataset? If so, please update the note here to be more human-readable. If not... no worries!

File: vaccine_feed_ingest/runners/nc/myspot_gov/normalize.py [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+            )
+        )
+    return ret
+
+
+def _get_organization(site: dict):
+    if site["Organization Name"] == "":
+        return None
+    if site["Organization Name"] == "Walmart, Inc.":
+        return schema.Organization(name=site["Organization Name"], id="walmart")
```
Comment (eliblock) [inline vaccine_feed_ingest/runners/nc/myspot_gov/normalize.py:97]: Thank you for doing this! I have [added it to the wiki](https://github.com/CAVaccineInventory/vaccine-feed-ingest/wiki/Normalized-Location-Schema#organization-enum)

**Review by eliblock — CHANGES_REQUESTED** [review]
Thank you for adding this normalizer! I fixed the git issue for you, and left a few suggestions before we land this.

File: vaccine_feed_ingest/runners/nc/myspot_gov/normalize.py (hunk 3 above)
Comment (rossry) [reply vaccine_feed_ingest/runners/nc/myspot_gov/normalize.py:103]: Means https://covid19.ncdhhs.gov/vaccines/providers/covid-19-vaccine-management-system-cvms -- I can get this into another note for these.

File: vaccine_feed_ingest/runners/nc/myspot_gov/normalize.py (hunk 1 above)
Comment (rossry) [reply vaccine_feed_ingest/runners/nc/myspot_gov/normalize.py:47]: On it; should be able to address in ~8 hours.

File: vaccine_feed_ingest/runners/nc/myspot_gov/normalize.py [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+            opens=_normalize_date(site["Start Date"]),
+            closes=_normalize_date(site["End Date"]),
+        )
+    ]
+
+
+def _get_inventories(site: dict):
+    ret = []
+    if site["Moderna"] == "Y":
+        ret.append(schema.Vaccine(vaccine="moderna", supply_level="in_stock"))
```
[the remark this answers is not in the document]
Comment (rossry) [reply vaccine_feed_ingest/runners/nc/myspot_gov/normalize.py:71]: Wow, I didn't realize. Very cool!

File: vaccine_feed_ingest/runners/nc/myspot_gov/normalize.py [hunk 6]
```diff
@@ -0,0 +1,151 @@
...
+        return None
+    if site["Organization Name"] == "Walmart, Inc.":
+        return schema.Organization(name=site["Organization Name"], id="walmart")
+    return schema.Organization(name=site["Organization Name"])
+
+
+def _get_notes(site: dict):
+    ret = []
+    ret.append("cvms_scheduling:" + site["CVMS Scheduling"])
+    ret.append("county:" + site["County"])
```
[the remark this answers is not in the document]
Comment (rossry) [reply vaccine_feed_ingest/runners/nc/myspot_gov/normalize.py:107]: It isn't explicitly in the address, but I guess it's implied. I'll drop it.

**Comment by rossry** [comment]
Resolved the requests. Here's newest output:

```{json}
{"id": "nc:myspot_gov:E-0000015", "name": "Kernersville Pharmacy", "address": {"street1": "841 Old Winston Road", "street2": "suite 90", "city": "Kernersville", "state": "NC", "zip": "27284"}, "location": {"latitude": 36.1122068, "longitude": -80.0905011}, "contact": [{"contact_type": null, "phone": "(336) 497-4511", "website": null, "email": null, "other": null}, {"contact_type": null, "phone": null, "website": "https://kernersvillepharmacy.com/", "email": null, "other": null}], "languages": null, "opening_dates": [{"opens": "2021-03-15", "closes": null}], "opening_hours": null, "availability": null, "inventory": [], "access": null, "parent_organization": {"id": null, "name": "Kernersville Pharmacy LLC"}, "links": null, "notes": ["cvms_scheduling__nc_specific:F", "cvms_info__nc_specific:https://covid19.ncdhhs.gov/vaccines/providers/covid-19-vaccine-management-system-cvms"], "active": null, "source": {"source": "nc:myspot_gov", "id": "E-0000015", "fetched_from_uri": "https://myspot.nc.gov/api/get-vaccine-locations", "fetched_at": "2021-04-27T12:21:58.432418", "published_at": null, "data": {"City": "Kernersville", "latitude": "36.1122068", "Postal Code": 27284, "State": "NC", "Start Date": "20210315 000000.000", "longitude": "-80.0905011", "Moderna": "", "Event Type": "Not Applicable", "Provider Location Name": "Kernersville Pharmacy", "Event Location Id": "E-0000015", "Salesforce Id": "a2qt000000lSlkLAAS", "Janssen": "", "Appointments": "Yes, Both", "Web Address": "https://kernersvillepharmacy.com/", "Organization Name": "Kernersville Pharmacy LLC", "Event Location": "", "Pfizer": "", "Street Address": "841 Old Winston Road", "County": "Forsyth", "CVMS Scheduling": "F", "Appointment Phone": "(336) 497-4511", "Display to the Public": "Y", "Street Address 2": "suite 90", "End Date": ""}}}
{"id": "nc:myspot_gov:E-0000357", "name": "Advance Community Health", "address": {"street1": "1011 Rock Quarry Road", "street2": "", "city": "Raleigh", "state": "NC", "zip": "27610"}, "location": {"latitude": 35.7670203, "longitude": -78.6157125}, "contact": [{"contact_type": null, "phone": "(919) 831-3111", "website": null, "email": null, "other": null}, {"contact_type": null, "phone": null, "website": "http://www.advancechc.org/vaccine/", "email": null, "other": null}], "languages": null, "opening_dates": [{"opens": "2021-03-15", "closes": null}], "opening_hours": null, "availability": null, "inventory": [{"vaccine": "moderna", "supply_level": "in_stock"}, {"vaccine": "pfizer_biontech", "supply_level": "out_of_stock"}, {"vaccine": "johnson_johnson_janssen", "supply_level": "out_of_stock"}], "access": null, "parent_organization": {"id": null, "name": "Advance Community Health, Inc."}, "links": null, "notes": ["cvms_scheduling__nc_specific:F", "cvms_info__nc_specific:https://covid19.ncdhhs.gov/vaccines/providers/covid-19-vaccine-management-system-cvms"], "active": null, "source": {"source": "nc:myspot_gov", "id": "E-0000357", "fetched_from_uri": "https://myspot.nc.gov/api/get-vaccine-locations", "fetched_at": "2021-04-27T12:21:58.432418", "published_at": null, "data": {"Pfizer": "N", "Appointments": "No, not taking appointments or walk-ins", "Event Type": "Not Applicable", "Street Address": "1011 Rock Quarry Road", "Start Date": "20210315 000000.000", "Organization Name": "Advance Community Health, Inc.", "Event Location": "", "CVMS Scheduling": "F", "State": "NC", "Janssen": "N", "Street Address 2": "", "End Date": "", "Event Location Id": "E-0000357", "Postal Code": "27610", "Web Address": "http://www.advancechc.org/vaccine/", "Provider Location Name": "Advance Community Health", "Salesforce Id": "a2qt000000lSlqQAAS", "Appointment Phone": "919-831-3111", "County": "Wake", "Display to the Public": "Y", "City": "Raleigh", "Moderna": "Y", "latitude": "35.7670203", "longitude": "-78.6157125"}}}
{"id": "nc:myspot_gov:E-0000358", "name": "Alamance County Health Department", "address": {"street1": "319 N Graham Hopedale Rd Suite B", "street2": "", "city": "Burlington", "state": "NC", "zip": "27217"}, "location": {"latitude": 36.1004866, "longitude": -79.4052922}, "contact": [{"contact_type": null, "phone": "(336) 290-0650", "website": null, "email": null, "other": null}, {"contact_type": null, "phone": null, "website": "https://www.alamance-nc.com/healthdept/", "email": null, "other": null}], "languages": null, "opening_dates": [{"opens": "2021-03-15", "closes": null}], "opening_hours": null, "availability": null, "inventory": [{"vaccine": "moderna", "supply_level": "in_stock"}, {"vaccine": "pfizer_biontech", "supply_level": "in_stock"}, {"vaccine": "johnson_johnson_janssen", "supply_level": "out_of_stock"}], "access": null, "parent_organization": {"id": null, "name": "Alamance County Health Department"}, "links": null, "notes": ["cvms_scheduling__nc_specific:F", "cvms_info__nc_specific:https://covid19.ncdhhs.gov/vaccines/providers/covid-19-vaccine-management-system-cvms"], "active": null, "source": {"source": "nc:myspot_gov", "id": "E-0000358", "fetched_from_uri": "https://myspot.nc.gov/api/get-vaccine-locations", "fetched_at": "2021-04-27T12:21:58.432418", "published_at": null, "data": {"Event Type": "Not Applicable", "longitude": "-79.4052922", "Street Address": "319 N Graham Hopedale Rd Suite B", "City": "Burlington", "Display to the Public": "Y", "Web Address": "https://www.alamance-nc.com/healthdept/", "latitude": "36.1004866", "Event Location": "", "Moderna": "Y", "Pfizer": "Y", "State": "NC", "Start Date": "20210315 000000.000", "CVMS Scheduling": "F", "County": "Alamance", "Postal Code": "27217", "End Date": "", "Janssen": "N", "Organization Name": "Alamance County Health Department", "Appointments": "Yes, Appointments Only", "Salesforce Id": "a2qt000000lSlqRAAS", "Appointment Phone": "336-290-0650", "Event Location Id": "E-0000358", "Street Address 2": "", "Provider Location Name": "Alamance County Health Department"}}}
{"id": "nc:myspot_gov:E-0000359", "name": "Chowan County Health Department", "address": {"street1": "202 West Hicks St", "street2": "", "city": "Edenton", "state": "NC", "zip": "27932"}, "location": {"latitude": 36.068496, "longitude": -76.609741}, "contact": [{"contact_type": null, "phone": "(252) 482-6003", "website": null, "email": null, "other": null}, {"contact_type": null, "phone": null, "website": "http://www.arhs-nc.org/", "email": null, "other": null}], "languages": null, "opening_dates": [{"opens": "2021-03-15", "closes": null}], "opening_hours": null, "availability": null, "inventory": [{"vaccine": "moderna", "supply_level": "in_stock"}, {"vaccine": "pfizer_biontech", "supply_level": "out_of_stock"}, {"vaccine": "johnson_johnson_janssen", "supply_level": "out_of_stock"}], "access": null, "parent_organization": {"id": null, "name": "Albemarle Regional Health Services"}, "links": null, "notes": ["cvms_scheduling__nc_specific:F", "cvms_info__nc_specific:https://covid19.ncdhhs.gov/vaccines/providers/covid-19-vaccine-management-system-cvms"], "active": null, "source": {"source": "nc:myspot_gov", "id": "E-0000359", "fetched_from_uri": "https://myspot.nc.gov/api/get-vaccine-locations", "fetched_at": "2021-04-27T12:21:58.432418", "published_at": null, "data": {"Appointments": "Yes, Walk-In Only", "Salesforce Id": "a2qt000000lSlqSAAS", "Provider Location Name": "Chowan County Health Department", "End Date": "", "Street Address 2": "", "latitude": "36.0684960", "Janssen": "N", "Event Location": "", "Moderna": "Y", "Organization Name": "Albemarle Regional Health Services", "Street Address": "202 West Hicks St", "Pfizer": "N", "longitude": "-76.6097410", "Event Type": "Not Applicable", "Postal Code": "27932", "Web Address": "http://www.arhs-nc.org/", "Event Location Id": "E-0000359", "City": "Edenton", "Appointment Phone": "(252) 482-6003", "County": "Chowan", "CVMS Scheduling": "F", "Start Date": "20210315 000000.000", "State": "NC", "Display to the Public": "Y"}}}
{"id": "nc:myspot_gov:E-0000360", "name": "Bertie County Health Department", "address": {"street1": "102 Rhodes Avenue", "street2": "", "city": "Windsor", "state": "NC", "zip": "27983"}, "location": {"latitude": 35.9891429, "longitude": -76.9301032}, "contact": [{"contact_type": null, "phone": "(252) 794-5322", "website": null, "email": null, "other": null}, {"contact_type": null, "phone": null, "website": "https://www.arhs-nc.org/", "email": null, "other": null}], "languages": null, "opening_dates": [{"opens": "2021-03-15", "closes": null}], "opening_hours": null, "availability": null, "inventory": [{"vaccine": "moderna", "supply_level": "in_stock"}, {"vaccine": "pfizer_biontech", "supply_level": "out_of_stock"}, {"vaccine": "johnson_johnson_janssen", "supply_level": "out_of_stock"}], "access": null, "parent_organization": {"id": null, "name": "Albemarle Regional Health Services"}, "links": null, "notes": ["cvms_scheduling__nc_specific:F", "cvms_info__nc_specific:https://covid19.ncdhhs.gov/vaccines/providers/covid-19-vaccine-management-system-cvms"], "active": null, "source": {"source": "nc:myspot_gov", "id": "E-0000360", "fetched_from_uri": "https://myspot.nc.gov/api/get-vaccine-locations", "fetched_at": "2021-04-27T12:21:58.432418", "published_at": null, "data": {"Event Location Id": "E-0000360", "Appointment Phone": "252-794-5322", "Display to the Public": "Y", "Salesforce Id": "a2qt000000lSlqTAAS", "Start Date": "20210315 000000.000", "Postal Code": "27983", "Web Address": "https://www.arhs-nc.org/", "Event Location": "", "latitude": "35.9891429", "End Date": "", "longitude": "-76.9301032", "State": "NC", "Street Address 2": "", "City": "Windsor", "County": "Bertie", "Appointments": "Yes, Walk-In Only", "Moderna": "Y", "Provider Location Name": "Bertie County Health Department", "Janssen": "N", "Street Address": "102 Rhodes Avenue", "Event Type": "Not Applicable", "Organization Name": "Albemarle Regional Health Services", "Pfizer": "N", "CVMS Scheduling": "F"}}}
{"id": "nc:myspot_gov:E-0000361", "name": "Gates County Health Department", "address": {"street1": "29 Medical Center Drive", "street2": "", "city": "Gates", "state": "NC", "zip": "27937"}, "location": {"latitude": 36.4525909, "longitude": -76.6999053}, "contact": [{"contact_type": null, "phone": "(252) 357-1380", "website": null, "email": null, "other": null}, {"contact_type": null, "phone": null, "website": "http://www.arhs-nc.org/", "email": null, "other": null}], "languages": null, "opening_dates": [{"opens": "2021-03-15", "closes": null}], "opening_hours": null, "availability": null, "inventory": [{"vaccine": "moderna", "supply_level": "in_stock"}, {"vaccine": "pfizer_biontech", "supply_level": "out_of_stock"}, {"vaccine": "johnson_johnson_janssen", "supply_level": "out_of_stock"}], "access": null, "parent_organization": {"id": null, "name": "Albemarle Regional Health Services"}, "links": null, "notes": ["cvms_scheduling__nc_specific:F", "cvms_info__nc_specific:https://covid19.ncdhhs.gov/vaccines/providers/covid-19-vaccine-management-system-cvms"], "active": null, "source": {"source": "nc:myspot_gov", "id": "E-0000361", "fetched_from_uri": "https://myspot.nc.gov/api/get-vaccine-locations", "fetched_at": "2021-04-27T12:21:58.432418", "published_at": null, "data": {"State": "NC", "Web Address": "http://www.arhs-nc.org/", "CVMS Scheduling": "F", "Start Date": "20210315 000000.000", "Provider Location Name": "Gates County Health Department", "City": "Gates", "Moderna": "Y", "Event Location Id": "E-0000361", "Display to the Public": "Y", "longitude": "-76.6999053", "Street Address": "29 Medical Center Drive", "End Date": "", "Appointment Phone": "252-357-1380", "Street Address 2": "", "latitude": "36.4525909", "Appointments": "Yes, Walk-In Only", "Event Location": "", "Pfizer": "N", "Postal Code": "27937", "Event Type": "Not Applicable", "Janssen": "N", "County": "Gates", "Organization Name": "Albemarle Regional Health Services", "Salesforce Id": "a2qt000000lSlqUAAS"}}}
```

File: vaccine_feed_ingest/runners/nc/myspot_gov/normalize.py (hunk 1 above)
Comment (rossry) [reply vaccine_feed_ingest/runners/nc/myspot_gov/normalize.py]: Did this.

Some phone numbers had additional info (like "extension 2" or "option 1"); I put this in the "other" field of the same contact entry, prepended with "phone_notes:". Let me know if we'd prefer something else?

Comment (eliblock) [reply vaccine_feed_ingest/runners/nc/myspot_gov/normalize.py]: That seems reasonable - our schema does not have a good way to handle options/extensions. CC: @bryanculbertson for potential modifications.

**Review by eliblock — APPROVED** [review]
Thank you for the update - I'm going to go ahead and merge! Excited to get this info onto the map soon!

Based on your later contributions it looks like you've figured out how to avoid having `il/sfsites` show up in every PR. If that's not the case - I am happy to help you get set up if those commands don't _quite_ work (as are the other folks maintaining this project). Let us know!

Looking forward to your next contribution, and thank you for the several so far.

**Pull request merged**